File: Vesti/Vesti/WebViewController.m
//
//  WebViewController.m
//  Vesti
//
//  Created by Djuro Alfirevic on 4/26/16.
//  Copyright © 2016 Djuro Alfirevic. All rights reserved.
//

#import "WebViewController.h"

@interface WebViewController() <UIWebViewDelegate>
@property (weak, nonatomic) IBOutlet UIWebView *webView;
@property (weak, nonatomic) IBOutlet UIActivityIndicatorView *spinnerView;
@end

@implementation WebViewController

#pragma mark - Actions

- (IBAction)backButtonTapped {
    [self dismissViewControllerAnimated:YES completion:NULL];
}

#pragma mark - View lifecycle

- (void)viewDidLoad {
    [super viewDidLoad];
    
    if (self.article) {
        NSURL *url = [NSURL URLWithString:self.article.link];
        NSURLRequest *request = [NSURLRequest requestWithURL:url];
        [self.webView loadRequest:request];
    }
}

#pragma mark - UIWebViewDelegate

- (void)webViewDidStartLoad:(UIWebView *)webView {
    [self.spinnerView startAnimating];
}

- (void)webViewDidFinishLoad:(UIWebView *)webView {
    [self.spinnerView stopAnimating];
}

@end

File: VestiS/VestiS/WebViewController.swift
//
//  WebViewController.swift
//  VestiS
//
//  Created by Djuro Alfirevic on 8/3/16.
//  Copyright © 2016 Djuro Alfirevic. All rights reserved.
//

import UIKit

class WebViewController: UIViewController, UIWebViewDelegate {

    // MARK: - Properties
    @IBOutlet weak var webView: UIWebView!
    @IBOutlet weak var spinnerView: UIActivityIndicatorView!
    var article: Article?
    
    // MARK: - Actions
    @IBAction func backButtonTapped(sender: UIButton) {
        self.dismissViewControllerAnimated(true, completion: nil)
    }
    
    // MARK: - View lifecycle
    override func viewDidLoad() {
        super.viewDidLoad()
        
        if let link = self.article?.link {
            if let url = NSURL(string: link) {
                let request = NSURLRequest(URL: url)
                self.webView.loadRequest(request)
            }
        }
    }
    
    // MARK: - UIWebViewDelegate
    func webViewDidStartLoad(webView: UIWebView) {
        self.spinnerView.startAnimating()
    }
    
    func webViewDidFinishLoad(webView: UIWebView) {
        self.spinnerView.stopAnimating()
    }

}